packages/tests/tests/connection_manager.spec.ts

```diff
@@ -34,162 +35,281 @@ describe("ConnectionManager", function () {
 
   afterEach(async () => {
     await waku.stop();
-    sinon.restore();
   });
 
-  describe("attemptDial method", function () {
-    let attemptDialSpy: SinonSpy;
+  describe("Events", () => {
+    it("should emit `peer:discovery:{bootstrap/peer-exchange}` event when a peer is discovered", async function () {
+      this.timeout(TEST_TIMEOUT);
 
-    beforeEach(function () {
-      attemptDialSpy = sinon.spy(connectionManager as any, "attemptDial");
-    });
+      const peerIdBootstrap = await createSecp256k1PeerId();
+
+      await waku.libp2p.peerStore.save(peerIdBootstrap, {
+        tags: {
+          [Tags.BOOTSTRAP]: {
+            value: 50,
+            ttl: 1200000,
+          },
+        },
+      });
+
+      const peerDiscoveryEventPromise = new Promise<void>((resolve) => {
+        connectionManager!.addListener(
+          PeerEvents.PEER_DISCOVERY_BOOTSTRAP,
+          (peerId) => {
+            expect(peerId.toString()).to.equal(peerIdBootstrap.toString());
+            resolve();
+          }
+        );
+      });
+
+      waku.libp2p.dispatchEvent(new CustomEvent("peer", { detail: peerId }));
+
+      await peerDiscoveryEventPromise;
 
-    afterEach(function () {
-      attemptDialSpy.restore();
+      const peerIdPx = await createSecp256k1PeerId();
+
+      await waku.libp2p.peerStore.save(peerIdPx, {
+        tags: {
+          [Tags.PEER_EXCHANGE]: {
+            value: 50,
+            ttl: 1200000,
+          },
+        },
+      });
+
+      const peerDiscoveryEventPromise2 = new Promise<void>((resolve) => {
+        connectionManager!.addListener(
+          PeerEvents.PEER_DISCOVERY_PEER_EXCHANGE,
+          (peerId) => {
+            expect(peerId.toString()).to.equal(peerIdPx.toString());
+            resolve();
+          }
+        );
+      });
+
+      waku.libp2p.dispatchEvent(new CustomEvent("peer", { detail: peerIdPx }));
+
+      await peerDiscoveryEventPromise2;
```

Comment:
I think we should have explicit `expect` in a test suit 

if you resolve from `peerDiscoveryBootstrap` and `peerDiscoveryPeerExchange` to `true` you can await them at the end of the test and set expectation `to.equal(true)` so that it is clear what the test is about